### PR TITLE
Use native `createRequireFromPath` function when available

### LINKED ISSUE #5: Use `module.createRequireFromPath()` when available
> Added module.createRequireFromPath(filename). This new method can be used to create a custom require function that will resolve modules relative to the filename path. #19360
> \- https://nodejs.org/en/blog/release/v10.12.0/

### PR DESCRIPTION
**Pull request by gillesdemey**
fixes #5

### CODE REVIEW

**Comment by gillesdemey** [comment]
Travis fails on Node 4 because `xo` is incompatible, should we just remove Node 4 from the version matrix?

File: index.js [hunk 1]
```diff
@@ -1,12 +1,24 @@
 'use strict';
+
+const Module = require('module');
+const path = require('path');
 const resolveFrom = require('resolve-from');
 
-module.exports = (fromDir, moduleId) => require(resolveFrom(fromDir, moduleId));
+const hasNativeFunction = typeof Module.createRequireFromPath === 'function';
+
+const requireFromPath = hasNativeFunction ?
```
Comment (kevva) [inline index.js]: This is a bit more readable:

```js
const requireFromPath = (fromDir, moduleId) => {
	if (!hasNativeFunction) {
		return require(resolveFrom(fromDir, moduleId));
	}

	return Module.createRequireFromPath(path.resolve(fromDir, 'index.js'))(moduleId);
};
```

File: index.js [hunk 2]
```diff
@@ -1,12 +1,24 @@
 'use strict';
+
```
Comment (kevva) [inline index.js]:
```suggestion

```

**Review by kevva — CHANGES_REQUESTED** [review]

**Comment by gillesdemey** [comment]
Done, PTAL :)

**Review by kevva — APPROVED** [review]

**Comment by sindresorhus** [comment]
`createRequireFromPath` is deprecated in favor of `createRequire`. Can you update to use that and also fix the merge conflict?

**Pull request closed without merging**